packages/web-components/fast-components/src/avatar/avatar.styles.ts

```diff
@@ -0,0 +1,68 @@
+import { css } from "@microsoft/fast-element";
+import { display } from "@microsoft/fast-foundation";
+import { accentFillRestBehavior, neutralForegroundRestBehavior } from "../styles/index";
+
+export const AvatarStyles = css`
+    ${display("flex")} :host {
+        max-width: var(--avatar-size, var(--avatar-size-default));
+        height: var(--avatar-size, var(--avatar-size-default));
+        position: relative;
+        --avatar-size-default: calc(
+            (
+                    (var(--base-height-multiplier) + var(--density)) * var(--design-unit) +
+                        (var(--design-unit) * 10)
+                ) * 1px
+        );
+        --avatar-text-size: var(--type-ramp-base-font-size);
+        --avatar-text-ratio: var(--design-unit);
+    }
+
+    .link {
+        text-decoration: none;
+        color: ${neutralForegroundRestBehavior.var};
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+        min-width: 100%;
+    }
+
+    .square {
+        border-radius: calc(var(--corner-radius) * 1px);
+        min-width: 100%;
+        overflow: hidden;
+    }
+
+    .coin {
+        position: relative;
+        display: flex;
+    }
+
+    .circle {
+        border-radius: 100%;
+        min-width: 100%;
+        overflow: hidden;
+    }
+
+    .image {
+        width: 100%;
+        position: absolute;
+        display: block;
+    }
+
+    .name {
+        font-size: calc(
```

Comment:
I think this is the only place this has come up, but I'm concerned designers won't want calculated font sizes that don't land on the ramp. Thinking through the algorithm, I think it would be something like "use the next smallest type ramp size where the line height is less than or equal to half the avatar height". I realize there's not a good way to handle that in css though. Wanted to propose it to see what solutions we might have.

Reply:
@bheston do you have additional thoughts or ideas on this?